mmcv/ops/point_sample.py

```diff
@@ -45,15 +45,16 @@ def bilinear_grid_sample(im, grid, align_corners=False):
     padded_w = w + 2
     # save points positions after padding
     x0, x1, y0, y1 = x0 + 1, x1 + 1, y0 + 1, y1 + 1
+
     # Clip coordinates to padded image size
-    x0[x0 < 0] = 0
-    x0[x0 > padded_w - 1] = padded_w - 1
-    x1[x1 < 0] = 0
-    x1[x1 > padded_w - 1] = padded_w - 1
-    y0[y0 < 0] = 0
-    y0[y0 > padded_h - 1] = padded_h - 1
-    y1[y1 < 0] = 0
-    y1[y1 > padded_h - 1] = padded_h - 1
+    x0 = (torch.clamp(x0.float() / (padded_w - 1), 0, 1) *
+          (padded_w - 1)).long()
+    x1 = (torch.clamp(x1.float() / (padded_w - 1), 0, 1) *
+          (padded_w - 1)).long()
+    y0 = (torch.clamp(y0.float() / (padded_h - 1), 0, 1) *
+          (padded_h - 1)).long()
+    y1 = (torch.clamp(y1.float() / (padded_h - 1), 0, 1) *
+          (padded_h - 1)).long()
```

Comment:
@RunningLeon, this fix leads to a metrics drop. My variation works with the same quality as custom ort op.

Reply:
This change is to support dynamic shape. If it results in a performance drop, you could post your tested results here.

Reply:
Kindly ping @DmitriySidnev

Reply:
@RunningLeon, I am collecting metrics in onnx runtime on cpu. It is very slow.

Reply:
@DmitriySidnev, Yes. It's quite slow with onnxruntime on cpu.

Reply:
My results with `clamp` function:
| Config                                      | Runtime | Bbox mAP | Mask mAP |
|---------------------------------------------|---------|----------|----------|
| point_rend_r50_caffe_fpn_mstrain_3x_coco.py | Pytorch | 41.0     | 38.0     |
| point_rend_r50_caffe_fpn_mstrain_3x_coco.py | ONNX    | 40.6     | 36.1     |

Here is a huge drop for segmentation mAP. Also I will collect metrics without `clamp` function to compare results.

Reply:
@DmitriySidnev,  Hi, if the performace of with `clamp` is worse than without `clamp`, please also post tested results here to verify this issue.

Reply:
Here is metrics without the `clamp` function (sorry for the delay, on cpu it works too slow):
| Config                                      | Runtime | Bbox mAP | Mask mAP |
|---------------------------------------------|---------|----------|----------|
| point_rend_r50_caffe_fpn_mstrain_3x_coco.py | ONNX    | 40.6     | 36.7     |

So, mAP for masks is better compared to the model with `clamp` function in python implementation of `grid_sampler`. But there is one more point from my side: presence of `clamp` function does not fix dynamic batch (nothing to fix). Dynamic batch works without it. I propose to retrieve my code.

Reply:
@DmitriySidnev, thanks for the effort of testing. Actually `clamp` is for dynamic shape with different input height and width.
If there's no need for dynamic, it could be replaced with the original style.

Reply:
@DmitriySidnev You may need to update the code accordingly.

Reply:
Done.